packages/create-cloudflare/src/templateMap.ts

```diff
@@ -1,53 +1,368 @@
+		([key, config]) => ({
+			label: config.displayName,
+			value: key,
+		})
+	);
+
+	const framework = await processArgument<string>(args, "framework", {
+		type: "select",
+		label: "framework",
+		question: "Which development framework do you want to use?",
+		options: frameworkOptions,
+		defaultValue: C3_DEFAULTS.framework,
+	});
+
+	framework || crash("A framework must be selected to continue.");
+	if (!Object.keys(frameworkMap).includes(framework)) {
+		crash(`Unsupported framework: ${framework}`);
+	}
+
+	const defaultFrameworkConfig = {
+		deployCommand: ["pages:deploy"],
+		devCommand: ["pages:dev"],
+	};
+
+	return {
+		...defaultFrameworkConfig,
+		...frameworkMap[framework as FrameworkName],
+	};
+};
+
+export async function copyTemplateFiles(ctx: C3Context) {
+	if (!ctx.template.copyFiles) {
+		return;
+	}
+
+	const { copyFiles } = ctx.template;
+
+	let srcdir;
+	if (copyFiles.path) {
+		// If there's only one variant, just use that.
+		srcdir = join(getTemplatePath(ctx), (copyFiles as VariantInfo).path);
+	} else {
+		// Otherwise, have the user select the one they want
+		const typescript = await shouldUseTs(ctx);
+		const languageTarget = typescript ? "ts" : "js";
+
+		const variantPath = (copyFiles as StaticFileMap)[languageTarget].path;
+		srcdir = join(getTemplatePath(ctx), variantPath);
+	}
+	const destdir = ctx.project.path;
+
+	const s = spinner();
+	s.start(`Copying template files`);
+
+	// copy template files
+	await cp(srcdir, destdir, { recursive: true, force: true });
+
+	// reverse renaming from build step
+	const dummyGitIgnorePath = join(destdir, "__dot__gitignore");
+	if (existsSync(dummyGitIgnorePath)) {
+		await rename(dummyGitIgnorePath, join(destdir, ".gitignore"));
+	}
+
+	s.stop(`${brandColor("files")} ${dim("copied to project directory")}`);
+}
+
+const shouldUseTs = async (ctx: C3Context) => {
+	// If we can infer from the directory that it uses typescript, use that
+	if (usesTypescript(ctx)) {
+		return true;
+	}
+
+	// Otherwise, prompt the user for their TS preference
+	return processArgument<boolean>(ctx.args, "ts", {
+		type: "confirm",
+		question: "Do you want to use TypeScript?",
+		label: "typescript",
+		defaultValue: C3_DEFAULTS.ts,
+	});
+};
+
+export const processRemoteTemplate = async (args: Partial<C3Args>) => {
+	const templateUrl = await processArgument<string>(args, "template", {
+		type: "text",
+		question:
+			"What's the url of git repo containing the template you'd like to use?",
+		label: "repository",
+		validate: validateTemplateUrl,
+		defaultValue: C3_DEFAULTS.template,
+	});
+
+	const path = await downloadRemoteTemplate(templateUrl);
+	const config = inferTemplateConfig(path);
+	validateTemplate(path, config);
+
+	return {
+		path,
+		...config,
+	};
+};
+
+const validateTemplate = (path: string, config: TemplateConfig) => {
+	if (config.copyFiles?.path) {
+		validateTemplateSrcDirectory(
+			resolve(path, config.copyFiles.path as string),
+			config
+		);
```

Comment:
what about:
```suggestion
	if (typeof config.copyFiles?.path === "string") {
		validateTemplateSrcDirectory(
			resolve(path, config.copyFiles.path),
			config
		);
```
and avoid casting?